openmc_plotter/plotmodel.py

```diff
@@ -501,19 +500,27 @@ def _do_op(array, tally_value, ax=0):
 
         return image_data, None, data_min, data_max
 
-    def _create_distribcell_image(self, tally, tally_value, scores, nuclides):
-        dfilter = tally.find_filter(openmc.DistribcellFilter)
-
+    def _create_distribcell_image(self, tally, tally_value, scores, nuclides, cellinstance=False):
+        # Get flattened array of tally results
         data = tally.get_values(scores=scores, nuclides=nuclides, value=tally_value)
         data = data.flatten()
 
-        cell_id = dfilter.bins[0]
-        # create a mask for ids that match the cell
+        # Create an empty array of appropriate shape for image
         image_data = np.full_like(self.ids, np.nan, dtype=float)
 
-        cell_id_mask = self.cell_ids == cell_id
-        for i, v in enumerate(data):
-            instance_mask = self.instances == i
+        # Determine appropriate set of bins depending on filter type
+        if cellinstance:
+            f = tally.find_filter(openmc.CellInstanceFilter)
+            bins = f.bins
+        else:
+            f = tally.find_filter(openmc.DistribcellFilter)
+            bins = [(f.bins[0], i) for i in range(data.size)]
+
+        # Iterate over tally bins, setting any pixels that have matching (cell
+        # ID, instance) each time
+        for v, (cell_id, instance) in zip(data, bins):
+            cell_id_mask = (self.cell_ids == cell_id)
```

Comment:
I'm not sure how much of a performance hit this might be, but having this in the loop is going to cause a repeated calculation of this mask for `DistribCellFilter`'s. The same is true for `CellInstanceFilter`'s that contain instances of the same cell. 

I suppose that since  we're typically working with around 100k pixels in the image data this shouldn't be too bad. Mainly noting it here in case we run into a performance issue with these filter types in the future.

Reply:
Yup, I'm well aware that this is performance-shitty code. But, so far on the models I've played with it doesn't seem to be a big deal. It would be easy enough to optimize if we find later on that it is causing significant lags on larger models. I can go ahead and add a comment in the code about this since I'm sure we'll probably forget.